src/core/core.controller.js

```diff
@@ -261,14 +281,29 @@ module.exports = function(Chart) {
 					scaleOptions.position = item.dposition;
 				}
 
-				var scale = new scaleClass({
-					id: scaleOptions.id,
-					options: scaleOptions,
-					ctx: me.ctx,
-					chart: me
+				var scale;
+				if (scales.hasOwnProperty(scaleOptions.id)) {
+					updated[scaleOptions.id] = true;
+					scale = scales[scaleOptions.id];
+					scale.options = scaleOptions;
+					scale.ctx = me.ctx;
+					scale.chart = me;
+				} else {
+					scale = new scaleClass({
+						id: scaleOptions.id,
+						options: scaleOptions,
+						ctx: me.ctx,
+						chart: me
+					});
+					scales[scale.id] = scale;
+				}
+				// clear up discarded scales
+				helpers.each(updated, function(id, hasUpdated) {
```

Comment:
I'm not sure that this loop is in the right place. I think it needs to happen after the entire outer loop has run. On the first iteration of the outer loop, only 1 scale will be updated and this will delete all the other axes